File: ThreadingSimulator/ThreadingSimulator/Models/VariableBlockModel.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Serialization;
using ThreadingSimulator.Enums;

namespace ThreadingSimulator.Models
{
    public class VariableBlockModel : BlockModel
    {
        [XmlIgnore]
        [JsonIgnore]
        private VariableOperationType variableOperationType;

        public string VariableName { get; set; }
        public VariableOperationType VariableOperationType
        {
            get
            {
                return variableOperationType;
            }
            set
            {
                variableOperationType = value;
                OnPropertyChanged();

                if(variableOperationType == VariableOperationType.GET)
                {
                    Value = null;
                    OnPropertyChanged(nameof(Value));
                }
            }
        }
        public int? Value { get; set; }

        [XmlIgnore]
        [JsonIgnore]
        public override bool IsValid
        {
            get
            {
                return !String.IsNullOrEmpty(VariableName) && (VariableOperationType == VariableOperationType.GET || Value!=null);
            }
        }

        public VariableBlockModel() : base(BlockType.SHARED_VARIABLE)
        {
            VariableOperationType = VariableOperationType.GET;
        }
    }
}

File: GUI/src/app/Models/VariableBlockModel.ts
import { BlockType } from "../Enums/BlockType";
import { VariableOperationType } from "../Enums/VariableOperationType";
import { isNullOrUndefined } from "util";
import { BlockModel } from "./BlockModel";

export class VariableBlockModel extends BlockModel
{
    public VariableOperationType: VariableOperationType = VariableOperationType.GET;

    public VariableName: string;
    
    public Value: number;

    public IsValid(): boolean
    {
        return !isNullOrUndefined(this.VariableName) && (this.VariableOperationType == VariableOperationType.GET || !isNullOrUndefined(this.Value));
    }
   
    public Clone(): VariableBlockModel
    {
        return VariableBlockModel.Deserialize(this);
    }

    public static Deserialize(obj: any): VariableBlockModel
    {
        let trueObj: VariableBlockModel = new VariableBlockModel();

        trueObj.Type = obj.Type;
        trueObj.Value = obj.Value;
        trueObj.VariableName = obj.VariableName;
        trueObj.VariableOperationType = obj.VariableOperationType;

        return trueObj;
    }

    constructor()
    {
        super(BlockType.SHARED_VARIABLE);
    }
}